packages/components/app/styles/components/side-nav/content.scss

```diff
@@ -147,8 +153,10 @@
 // LIST ITEM > INNER ELEMENTS
 
 .hds-side-nav__list-item-text {
+  max-width: 100%;
```

Comment:
We need to set a `min-width: 0` to force it to stay within its flex container
<img width="377" alt="modified" src="https://github.com/hashicorp/design-system/assets/686239/8f583554-771c-4691-8ab4-5d09f25f0fee">

```suggestion
  max-width: 100%;
  min-width: 0;
```

Reply:
Adding locally, thanks